src/flowchem/devices/knauer/knauer_autosampler_component.py

```diff
@@ -0,0 +1,315 @@
+        """Initialize component."""
+        super().__init__(name, hw_device)
+        self.add_api_route("/set_needle_position", self.set_needle_position, methods=["PUT"])
+
+
+    async def set_needle_position(self, position: str = "") -> None:
+        """
+        Move the needle to one of the predefined positions.
+
+        Argument:
+            position (str):
+                        WASH
+                        WASTE
+                        EXCHANGE
+                        TRANSPORT
+        """
+        if position != "PLATE":
+            await self.hw_device._move_needle_vertical(NeedleVerticalPositions.UP.name)
+            await self.hw_device._move_needle_horizontal(needle_position=position)
+        else:
+            raise NotImplementedError
+
+    async def set_xy_position(self, plate: str = "", row: int = 0, column: int = 0) -> None:
+        """
+        Move the 3D gantry to the specified (x, y) coordinate of a specific plate.
+
+        plate (str):
+                    NO_PLATE
+                    LEFT_PLATE
+                    RIGHT_PLATE
+                    SINGLE_PLATE
+
+        column (int): starting from 1.
+        row (int) starting from 1.
+        """
+        if await self.hw_device.get_status() == "NEEDLE_RUNNING":
+            logger.warning("Needle already moving!")
+
+        traytype = self.hw_device.tray_type.upper()
+        await self.hw_device._move_needle_vertical(NeedleVerticalPositions.UP.name)
+        if traytype in PlateTypes.__dict__.keys():
+            try:
+                if PlateTypes[traytype] == PlateTypes.SINGLE_TRAY_87:
+                    raise NotImplementedError
+            except KeyError as e:
+                raise Exception(
+                    f"Please provide one of following plate types: {[i.name for i in PlateTypes]}") from e
+            # now check if that works for selected tray:
+            assert PlateTypes[traytype].value[0] >= column and PlateTypes[traytype].value[1] >= row
+            await self.hw_device._move_tray(plate, row)
+            success = await self.hw_device._move_needle_horizontal(NeedleHorizontalPosition.PLATE.name, plate=plate, well=column)
+            if success:
+                logger.info("Needle moved successfully to row: {row}, column: {column} on plate: {plate}")
+            return
+        else:
+            raise NotImplementedError
+
+    async def set_z_position(self, direction: str = "") -> None:
+        """
+        Move the 3D gantry along the Z axis.
+
+        Argument:
+            direction (str):
+            DOWN
+            UP
+        """
+        if await self.hw_device.get_status() == "NEEDLE_RUNNING":
+            logger.warning("Needle already moving!")
+
+        success = await self.hw_device._move_needle_vertical(move_to=direction)
+        if success:
+            logger.info("Needle moved successfully to {direction} direction.")
+
+
+    #ToDo
```

Comment:
I didn't find a command to query the current vertical position, as there is for querying the valve's position for example.